File: frontend/src/types/status.ts
import type { EntityConfig } from './domain';

export type DefectState = 'new' | 'verified' | 'monitoring' | 'mitigated' | 'closed';
export const ALL_DEFECT_STATE: readonly DefectState[] = ['new', 'verified', 'monitoring', 'mitigated', 'closed'];
export type PriorityLevel = 'observe' | 'restrict' | 'urgent';
export const ALL_PRIORITY_LEVEL: readonly PriorityLevel[] = ['observe', 'restrict', 'urgent'];
export type DispositionLevel = 'observe' | 'restrict' | 'urgent';
export const ALL_DISPOSITION_LEVEL: readonly DispositionLevel[] = ['observe', 'restrict', 'urgent'];
export type DefectGrade = 'general' | 'serious';
export const DISPOSITION_LEVEL_LABELS: Record<DispositionLevel, string> = {
  observe: '观察',
  restrict: '限行管控',
  urgent: '立即处置',
};
export const DEFECT_GRADE_LABELS: Record<DefectGrade, string> = {
  general: '一般',
  serious: '严重',
};

export const ENTITY_CONFIGS: readonly EntityConfig[] = [
  { key: 'bridgeAsset', path: 'bridges', label: '桥梁资产', statuses: ['active', 'restricted', 'closed', 'retired'] as const },
  { key: 'inspectionRound', path: 'inspections', label: '检查批次', statuses: ['planned', 'running', 'review', 'completed'] as const },
  { key: 'defectFinding', path: 'defects', label: '缺陷发现', statuses: ['new', 'verified', 'monitoring', 'mitigated', 'closed'] as const },
  { key: 'priorityDecision', path: 'priorities', label: '优先级决定', statuses: ['draft', 'observe', 'restrict', 'urgent'] as const }
];

export const DISPOSITION_PAGE_CONFIG = { key: 'dispositionAdvice', path: 'disposition-advices', label: '处置建议' } as const;

File: backend/internal/constants/status.go
package constants

// Shared status values are mirrored in frontend/src/types/status.ts. Keeping
// the lists explicit makes state-machine drift visible during code review.

type DefectState string

const (
	DefectStateNew        DefectState = "new"
	DefectStateVerified   DefectState = "verified"
	DefectStateMonitoring DefectState = "monitoring"
	DefectStateMitigated  DefectState = "mitigated"
	DefectStateClosed     DefectState = "closed"
)

var AllDefectState = []string{"new", "verified", "monitoring", "mitigated", "closed"}

type PriorityLevel string

const (
	PriorityLevelObserve  PriorityLevel = "observe"
	PriorityLevelRestrict PriorityLevel = "restrict"
	PriorityLevelUrgent   PriorityLevel = "urgent"
)

var AllPriorityLevel = []string{"observe", "restrict", "urgent"}

// DispositionAdviceLifecycleState 缺陷处置优先级建议只有一个生命周期状态：
// 生成即定稿（与快照同一事务写入），重复核验不产生第二份建议。
type DispositionAdviceLifecycleState string

const (
	DispositionAdviceStateGenerated DispositionAdviceLifecycleState = "generated"
)

var AllDispositionAdviceState = []string{"generated"}

// DefectGradeClass 按缺陷等级划分的处置档位来源：严重缺陷可被桥梁限行升级，
// 一般缺陷始终保持观察。
type DefectGradeClass string

const (
	DefectGradeGeneral DefectGradeClass = "general"
	DefectGradeSerious DefectGradeClass = "serious"
)

// RiskLevelToDefectGrade 将平台统一的风险等级映射为处置判定使用的缺陷等级。
func RiskLevelToDefectGrade(riskLevel string) DefectGradeClass {
	switch riskLevel {
	case "high", "critical":
		return DefectGradeSerious
	default:
		return DefectGradeGeneral
	}
}

var BridgeAssetTransitions = map[string]map[string]bool{
	"active":     {"restricted": true, "closed": true},
	"restricted": {"closed": true, "retired": true, "active": true},
	"closed":     {"retired": true, "restricted": true},
	"retired":    {"closed": true},
}

var InspectionRoundTransitions = map[string]map[string]bool{
	"planned":   {"running": true, "review": true},
	"running":   {"review": true, "completed": true, "planned": true},
	"review":    {"completed": true, "running": true},
	"completed": {"review": true},
}

var DefectFindingTransitions = map[string]map[string]bool{
	"new":        {"verified": true, "monitoring": true},
	"verified":   {"monitoring": true, "mitigated": true, "new": true},
	"monitoring": {"mitigated": true, "closed": true, "verified": true},
	"mitigated":  {"closed": true, "monitoring": true},
	"closed":     {"mitigated": true},
}

var PriorityDecisionTransitions = map[string]map[string]bool{
	"draft":    {"observe": true, "restrict": true, "urgent": true},
	"observe":  {},
	"restrict": {},
	"urgent":   {},
}

func CanTransition(graph map[string]map[string]bool, from, to string) bool {
	targets, exists := graph[from]
	return exists && targets[to]
}
